### ISSUE TITLE
BListen is obsolete

### ISSUE THREAD

**Issue by nickng** [rhu1/scribble-go-runtime#16]
The conversion is better done implicitly at function call site. The typical pattern is functions return struct but accept interface as parameter.

**Comment by rhu1** [comment]
It seems difficult to factor out transport-independent code involving listeners without BListen (this was why it was hacked in).  E.g., it's used here: 

https://github.com/nickng/scribble-go-examples/blob/rhu1-branch/scributil/scributil.go#L27

At least, that was my observation at that time.  Not sure if things have since evolved such that this comment is now obsolete.

**Comment by nickng** [comment]
You are right that something like `BListen` is needed for transport independence.

I just think that BListen does not belong in the concrete transport implementation packages but should exist as a generic `Listen` function in the transport package (acts as a switchboard to all transport implementations' `Listen` function). This design is used in [net.Listen](https://golang.org/pkg/net/#Listen), where you do:

    // net.Listen(network string, address string) (Listener, error)
    net.Listen("tcp", "0.0.0.0") return (*TCPListener, error)
    net.Listen("udp", 0) // returns (*UDPListener, error)

We could have something similar to what is in scributi

    // transport.Listen(transport string, port int) (Listener, error)
    ln, error := transport.Listen("tcp", 8888)
    ln, error := transport.Listen("shm", 10)
